doc/OPTIONS

```diff
@@ -394,29 +394,41 @@ their number could potentially be decreasing as some hashes get cracked.
-autodetect by default.  Mainly for test and list purposes (see --list=WHAT),
-you can use group aliases "dynamic", "cpu", "gpu", "opencl" and "cuda"
-as a format name, or use one wildcard, as in "--format=mysql-*",
-"--format=raw*ng" or "--format=*office".  There's also a special way
-to match any substring within the "algorithm name" (the string shown
-within brackets, e.g. "[DES 128/128 AVX-16]" using @tag syntax as in
-e.g. "--format=@DES" or "--format=@AVX".  Similarly you can match a
-substring in the "format name" (the longer name, not the label) using
-#tag format.  For example, if you forgot which format handles the IPMI
-hashes, you can say --format=#ipmi or --format=#ipmi-opencl and the
-correct format (in this case RAKP or RAKP-opencl) will be used.
-Regardless of method, the matching ignores case.
+--format=NAME[,NAME...]		force hash type NAME
+
+Override the hash type auto-detection.  You can use this option when
+you're starting a cracking session or along with things like: "--test",
+"--show", "-list" or "--make-charset".  Note that John can't crack hashes
+of different types at the same time.  If you happen to get a password file
+that uses more than one hash type, then you have to invoke John once for
+each hash type and you need to use this option to make John crack hashes of
+types other than the one it would autodetect by default.  For normal use,
+you'd use a full exact format label such as "--format=crypt-md5".  Mainly
+for test and list purposes (see --list=WHAT), you can use a single wildcard,
+as in "--format=mysql-*", "--format=raw*ng" or "--format=*office", or group
+aliases such as "dynamic", "cpu", "omp", "gpu", "opencl", "ztex" as a format
+name.  There's also a special way to match any substring within the
+"algorithm name" (the string shown within brackets, e.g. "[DES 128/128
+AVX-16]") using @tag syntax as in e.g. "--format=@DES" or "--format=@AVX".
+Similarly you can match a substring in the "format name" (the longer name,
+not the label) using #tag format.  For example, if you forgot which format
+handles the IPMI hashes, you can say --format=#ipmi and the correct format
+(in this case RAKP) will be used.  Regardless of method, the matching
+ignores case.  Several formats can be specified at once, separated by
+commas, and prepending a format, wildcard or class with a "-" will exclude
+instead of include (eg. "--format=#ipmi,-cpu" for again finding the RAKP
+format but this time rejecting the CPU version so will end up finding
+"RAKP-OpenCL" as long as your build supports OpenCL).  The order given of
+any "include" parts of a format list will be honored so this it feature can
+also be used for merely changing format precedence during auto-detection.
+Prepending a format, wildcard or class with a "+" will promote it to a
+requirement as opposed to an inclusion. That is, "--format=omp,+cpu"
+will first "include" any OMP formats but then prune all of them that does
+not match format class "CPU".
```

Comment:
This wall of text should be broken in paragraphs, but the rest of the file isn't made like that